### PR TITLE
Fix workspace URI long names when a workspace is a prefix of filename, but not actually parent

### PR DESCRIPTION
**Pull request by simark**
See commit messages for details.

### CODE REVIEW

File: packages/workspace/src/browser/workspace-uri-contribution.ts [hunk 1]
```diff
@@ -99,10 +102,6 @@ export class WorkspaceUriLabelProviderContribution extends DefaultUriLabelProvid
             return super.getLongName(uri);
         }
 
-        const short = uriStr.substr(this.wsRoot.length);
-        if (short[0] === '/') {
-            return short.substr(1);
-        }
-        return short;
+        return uriStr.substr(this.wsRoot.length);
```
Comment (akosyakov) [inline packages/workspace/src/browser/workspace-uri-contribution.ts]: How about getting a relative path: https://github.com/theia-ide/theia/blob/41b4d633ac04668a9e8c7068373a1e6494a5a66f/packages/core/src/common/uri.ts#L65

It is already well tested.

Comment (simark) [reply packages/workspace/src/browser/workspace-uri-contribution.ts]: Makes sense.  I updated the PR.

**Comment by akosyakov** [comment]
@simark Can you have at failed builds please?

**Comment by simark** [comment]
> @simark Can you have a look at failed builds, please?

Oops, done.

**Pull request merged**